providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/common/ChannelMonitor.java

```diff
@@ -47,8 +47,10 @@ public static void monitorChannelState(
                     log.debug("onConnectionLost is null");
                 }
             }
-            // Re-register the state monitor to watch for the next state transition.
-            monitorChannelState(currentState, channel, onConnectionReady, onConnectionLost);
+            if (currentState != ConnectivityState.SHUTDOWN) {
```

Comment:
shutdown means to end, I do not think we need to continue to monitor afterward; this might also be the reason why we sometimes fail in our tests to clean up some channels. just and idea